Sophia Sun Project Pitch

```diff
@@ -0,0 +1,244 @@
+
+The program asks the user to submit a txt file with the list of chords, each chord on one line of the txt file. 
+
+# What kinds of semantic (i.e., non-syntax) errors might occur, and how might they be communicated to the user?_
+
+There are ways to express a chord that may be outside of what I'm programming my parser to be able to understand. For example,
+I am parsing "Maj" and "Major" as a major chord. However, it's possible that a non-standard input might be "MAJOR" or "MAJ" or
+"Maj." My program could suggest a few possible key words that can replace the input, and tell the user which line threw an error.
+
+It's also possible that the chord sheet proposes a chord that doesn't exist (typo). The user might copy/paste
+in the name of a chord such as T Major with an inversion of 6/2. In this case, the program doesn't recognize T as a 
+possible root, and it would return that in an error message.
+
+### Expressiveness
+# What should be easy to do in this language?
+What should be really easy is to generate at least the names of all the notes that comprise a given chord that the user specifies.
+
+# What should be possible, but difficult? 
+What should be possible, but difficult, is to add a sound component to the program. 
+After the program determines the notes in a chord, presents a visual representation of the chord on a musical staff, 
+it might also have the functionality of playing the chord, note by note from the bottom note to the top note. 
+
+# What should be impossible or very difficult?
+
+What should be impossible is something outside the domain of generating the duration (length) of each chord. 
+There will be no opportunity for the user to input this sort of information.
+
+## Related work
+_Are there any other DSLs in this domain? If not, describe how you know there
+aren't and conjecture why not. If so, describe them and provide links. How well 
+do they address the need? Are there any particularly admirable qualities of the
+language? Are there parts of the language you think could be improved?_
+
+I searched google and scoured various composing tool threads with the following key phrases
+“chord name sheet music converter", "generate sheet music", "generate chords on sheet music",
+and didn’t find any existing DSL that is similar to my idea.
+
+This could be because most people who play accompaniment music play the guitar -- 
+there actually exists many "chord identifier" / "reverse chord finder" tools that prompt users to select 
+fingering positions on a guitar and then show users what possible chords are comprised of those notes. 
+
+
+### Suitability
+_What percentage of your time do you think will be spent directly engaging in
+the **language** aspects of this project (e.g., making language design
+decisions), as opposed to "systems" aspects of the project (e.g., implementing a
+complicated semantics that doesn't require a lot of language design)?_
+
+I don't think there will be a lot of language decisions -- 
```

Comment:
If you let the users enter the duration of chords, there would be more design decisions for you to make. I think this may be an important component to your DSL. You said in class that the program would produce sheet music with only quarter notes, I'm not sure if people who sight read can tell their brains to ignore information on duration and only read the notes. So it might be helpful to require duration of notes to be entered, even if it requires more work for the user.

Possible design problems could be whether the users would enter duration as seconds, which would be pretty accessible; or duration relative to a measure with defined time, which would require more domain expertise but would translate better to your program.